src/msftoolbox/reliefweb/reports.py

```diff
@@ -0,0 +1,253 @@
+            preset (str, optional): A shorthand specification of sets of fields, filters and sort order for common use-cases.
+                                    Defaults to "latest", which sorts by date for appropriate content types. Countries and sources sorted by id. .
+            limit (int, optional): How many results to return. Defaults to 1000, max 1000.
+            profile (str, optional): A shorthand specification for which sets of fields to include in result.. Defaults to "full".
+        """
+        self.app_name = app_name
+        self.preset = preset
+        self.limit = limit
+        self.profile = profile
+
+    def try_key(
+        self,
+        key:str,
+        data:dict
+        ):
+        """Tests a key in a dictionary and resolves to None if missing
+
+        Args:
+            key (str): They key to test for
+            data (dict): The dictionary to test in
+
+        Returns:
+            any: The value associated with the key
+        """
+        if key in data.keys():
+            return data[key]
+        else:
+            print(f"Key {key} missing in article {data['title']}")
+            return None
+
+    def validate_date(
+        self,
+        date_string:str
+        ) -> bool:
+        """Returns a boolean indicating if the date string is in "%Y-%m-%d" format
+
+        Args:
+            date_string (str): The date as a string
+
+        Returns:
+            bool: True is a correctly formatted string and False is a wrongly formatted string
+        """
+        try:
+            # Try to create a datetime object from the date_str
+            # This checks both the format and the validity of the date (e.g., no February 30)
+            datetime.strptime(date_string, "%Y-%m-%d")
+            return True
+        except ValueError:
+            # If ValueError is raised, the format is wrong or the date is invalid
+            return False
+
+
+    def list_reports(
+        self,
+        start_date:str,
+        end_date:str,
+        query_value:str,
+        query_fields:list = ["body", "title"],
+        query_operator:str = "OR",
+        country_filter:list = None,
+        response_format:str = "dictionary"
+        ) -> list:
+        """List all reports matching a query with optional filters.
+
+        Args:
+            start_date (str): The start date in YYYY-MM-DD format.
+            end_date (str): The end date in YYYY-MM-DD format.
+            query_value (str): The value to search for. This is equivalent to what would be entered in the Relief Web search bar.
+            query_fields (list): The fields to search in. Allowed values can be found at: https://apidoc.reliefweb.int/fields-tables.
+            query_operator (str): The operator ('OR', 'AND') determining how to treat queries with multiple search keywords.
+            country_filter (list): A list of ISO3 country codes for filtering. ISO3 codes can be found at: https://www.iban.com/country-codes.
```

Comment:
Perhaps would be better to have the country english names rather than ISO? The ISO3 country codes were nice because we had this mapping table which worked for RW, Gdelt, etc

Reply:
Yes I agree to an exten, but we would need to manage this mapping table and going against the inputs of the actual api calls you would need to maintain that then. E.g what if the Netherlands would like to rename itself to "derekland" then we would need to maintain that, I reference a resource to find the the ISO codes in the docstrings